src/Zip/ZipFile.Selector.cs

```diff
@@ -1255,8 +1255,8 @@ internal partial class NameCriterion : SelectionCriterion
     {
         internal override bool Evaluate(Ionic.Zip.ZipEntry entry)
         {
-            // swap forward slashes in the entry.FileName for backslashes
-            string transformedFileName = entry.FileName.Replace("/", "\\");
+            // swap slashes in reference to local configuration
+            string transformedFileName = entry.FileName.Replace(Path.DirectorySeparatorChar == '/' ? '\\' : '/', Path.DirectorySeparatorChar);

```

Comment:
No. I replace \ with / on Unix and / with \ on Windows. That's an inline boolean. Check here: https://msdn.microsoft.com/en-us/library/ty67wk28.aspx